bag.hpp

```diff
@@ -74,6 +74,13 @@ enum : int {
 		   printf("--\n");
 	   }
 
+
+	   void clear() {
+	       indices.fill(EMPTY);
+	       elements.fill(EMPTY);
+	       size_ = 0;
+	   }
+
```

Comment:
these lines can be deleted — at the moment there is no use case for spontaneously emptying a Bag, since we only ever construct one Universe. If the functionality turns out to be useful at a later stage, we can add the function back in. Until then, we'd best leave the core structure unchanged.